src/metabase/api/cards.clj

```diff
@@ -0,0 +1,53 @@
+(ns metabase.api.cards
+  "Bulk endpoints for Cards"
+  (:require [compojure.core :refer [POST]]
+            [metabase.api.card :as api.card]
+            [metabase.api.common :as api]
+            [metabase.models.interface :as mi]
+            [metabase.util.malli :as mu]
+            [metabase.util.malli.schema :as ms]
+            [toucan2.core :as t2]))
+
+(mu/defn get-dashboards
+  "Get dashboards that a card appears in. The result looks like
+  ```
+  {:card_id 123
+   :dashboards [{:error :unwritable-dashboard :name \"foo\" :id 456}
+                {:error :unreadable-dashboard}
+                {:name \"foo\" :id 678}]}
+  ```"
+  [card-id :- ms/PositiveInt]
+  (let [card (t2/select-one :model/Card :id card-id)
+        dashboards (:in_dashboards (t2/hydrate card :in_dashboards))]
+    {:card_id card-id :dashboards (->> dashboards
+                                       (map #(cond
+                                               (not (mi/can-read? %)) {:error :unreadable-dashboard}
+                                               (not (mi/can-write? %)) (merge % {:error :unwritable-dashboard})
+                                               :else %))
+                                       (map #(dissoc % :collection_id)))}))
+
+(api/defendpoint POST "/dashboards"
+  "Get the dashboards that multiple cards appear in. The response is a sequence of maps, each of which has a `card_id`
+  and `dashboards`. `dashboard` may include an `:error` key, either `:unreadable-dashboard` or
+  `:unwritable-dashboard`. In the case of an `unreadable-dashboard` the dashboard details (name, ID) will NOT be
+  present."
```

Comment:
For my own edification, what's the purpose of the `:unreadable-dashboard` markers?

Reply:
The idea is just to be able to tell the user "this card appears in dashboards you don't have the necessary permissions on", providing the names/ids where possible. It does feel a little weird to give them a list of identical `{:error :unreadable-dashboard}` maps. Maybe we can revisit this after merge to make this a bit more sane? Maybe the backend itself should be responsible for telling the frontend whether it could move a card and why, rather than the frontend needing to know about the fact that moving a card will depend on the permissions of all the dashboards it's in... 🤔 are you ok merging as is with a plan to revisit this soon though?

Reply:
Yeah, I think that's fine